buildsystem: improve maintainer information handling

feckert opened the pull request:
The two commits are not dependent on each other, but affect the maintainer handling.

- @aparcar remove duplicate MAINTAINER from package-dumpinfo.mk
- @Ansuel @ynezz remove Maintainer info from ipkg control file on the target

feckert committed:
build: remove duplicate MAINTAINER from package-dumpinfo.mk

This is an artifact and is not needed. We have already set the
information 'Maintainer:' in the '.packageinfo-*' files.

Signed-off-by: Florian Eckert <fe@dev.tdt.de>

feckert committed:
build: remove Maintainer info from ipkg control file

This information is not necessary/needed on the target and and can
therefore be removed from the control files during rootfs building.

Signed-off-by: Florian Eckert <fe@dev.tdt.de>

feckert commented:
@musashino205 thanks done

aparcar commented on config/Config-build.in:
```diff
@@ -114,6 +114,14 @@ menu "Global build settings"
+		bool
+		prompt "Remove packages maitainer information on the target"
+		help
+		  Attackers know, that developers have access to source code of the device
```
Don't you think attackers also know, that the source code contain maintainer information? If you try to malicoiously get commit access to source code aka a repository, you likely look at it before (e.g. to understand its impact). In that case one easily obtain that information anyway.

feckert replied:
That's not quite right. This is not about the creator of the source code but about the maintainer of the package at OpenWrt.
I know that you can get the information in other ways, but you don't have to make it too easy?
The creator of the OpenWrt package usually does not know the internals of the upstream package he maintains, but he can add patches, which is an attack vector if you put the OpenWrt package maintainer under pressure.

aparcar replied:
Sounds a bit like "security by obscurity" to me. Distributions make these information very easy accessible, like [Debian](https://packages.debian.org/sid/devel/tmate), [Alpine](https://pkgs.alpinelinux.org/package/edge/testing/x86_64/tmate) or even [OpenWrt](https://openwrt.org/packages/pkgdata/tmate).

feckert replied:
Thanks for your feedback. Absolutely right, I agree with you that is "security by obscurity" if that's what you want to call it. The original question from my side was actually do we need the information on the rootfs target?
Isn't it enough that we have it in the repo? Why not always remove it on the rootfs target?

aparcar commented on include/package-dumpinfo.mk:
```diff
@@ -44,7 +44,6 @@ $(if $(KCONFIG),Kernel-Config: $(KCONFIG)
 )$(if $(BUILDONLY),Build-Only: $(BUILDONLY)
 )$(if $(HIDDEN),Hidden: $(HIDDEN)
 )Description: $(if $(Package/$(1)/description),$(Package/$(1)/description),$(TITLE))
-$(MAINTAINER)
```
This looks fine to me.

feckert committed:
build: also generate the CycloneDX SBOM in imagebuilder

The needed file '.packageinfo' for creating the CycloneDX SBOM in the
imagebuilder is available in the top directory of the imagebuilder and
not in the tmp directory.

For this reason, the creation of the CycloneDX SBOM file is not available
for the imagebuilder.

To fix this, it is now first checked whether the CycloneDX SBOM should be
built at all and then second decided by checking the IB variable where the
'.packageinfo' file is to be found.

With this change, it is now possible to create the CycloneDX SBOM also for
the imagebuilder as well.

Signed-off-by: Florian Eckert <fe@dev.tdt.de>

feckert commented:
Thanks for the feedback.

aparcar commented on include/rootfs.mk:
```diff
@@ -60,6 +60,12 @@ ifdef CONFIG_CLEAN_IPKG
+# Remove not needed information from the control file
+# 1: roofs base path
+define clean_control_file
+	-sed -i '/^Maintainer: /d' $(1)/usr/lib/opkg/info/*.control
```
Please be aware that we eventually plan your switch to APK which renders those changes obsolete. Just a comment

feckert replied:
Thanks for the hint.
If this is the case then I will discard the newly opened PR this commit, that is removing this Information from the control file. What information is stored on the rootfs target when using APK? Is there any information on this yet?
